src/bin/opensmt.cc

```diff
@@ -238,4 +210,83 @@ void catcher( int sig )
   }
 }
 
+
+void printHelp(int code = 0)
+{
+    const char help_string[]
+        = "Usage: ./opensmt [OPTION] <filename>\n"
+          "where OPTION can be\n"
+          "  --help [-h]                     prints this help message and exits\n"
+          "  --version                       prints opensmt version and exits\n"
+          "  --dry-run [-d]                  executes dry run\n"
+          "  --random-seed [-r] <seed>       sets random seed to specific number\n"
+          "  --produce-interpolants [-i]     enables interpolant computation\n"
+          "  --verbose [-v]                  verbose run of the solver\n"
+          "  --pipe [-p]                     for execution within a pipe\n";
+    std::cerr << help_string;
+    exit(code);
+}
+
+SMTConfig parseCMDLineArgs( int argc, char * argv[ ] )
+{
+    SMTConfig res;
+    int versionOpt = 0;
+    int dryRunOpt = 0;
+
+    struct option long_options[] =
+        {
+            {"help", no_argument, nullptr, 'h'},
+            {"version", no_argument, &versionOpt, 1},
+            {"verbose", no_argument, nullptr, 'v'},
+            {"dry-run", no_argument, &dryRunOpt, 'd'},
```

Comment:
The long option `--dry-run` doesn't work.
```
% ./opensmt --dry-run
Usage: ./opensmt [OPTION] <filename>
where OPTION can be
  --help [-h]                     prints this help message and exits
  --version                       prints opensmt version and exits
  --dry-run [-d]                  executes dry run
  --random-seed [-r] <seed>       sets random seed to specific number
  --produce-interpolants [-i]     enables interpolant computation
  --verbose [-v]                  verbose run of the solver
  --pipe [-p]                     for execution within a pipe
```

The short one works:
```
% ./opensmt -d
(set-logic QF_UF)
(check-sat)
unknown
```

Probably related to what @Tomaqa writes [here](https://github.com/usi-verification-and-security/opensmt/pull/801/files#r1836457954).

Reply:
Fixed